clients/drcachesim/scheduler/scheduler_impl.h

```diff
@@ -195,8 +195,27 @@ template <typename RecordType, typename ReaderType> class scheduler_impl_tmpl_t
         // If non-empty these records should be returned before incrementing the reader.
         // This is used for read-ahead and inserting synthetic records.
         // We use a deque so we can iterate over it.
-        std::deque<RecordType> queue;
+        struct queued_record_t {
+            queued_record_t()
+            {
+            }
+            queued_record_t(const RecordType &record)
+                : record(record)
+            {
+            }
+            queued_record_t(const RecordType &record, uint64_t next_trace_pc)
+                : record(record)
+                , next_trace_pc(next_trace_pc)
+            {
+            }
+
+            RecordType record;
+            uint64_t next_trace_pc = 0;
```

Comment:
I don't understand exactly what this is: if 0 doesn't mean an error, is it that this is really the actual PC of this record itself and not the "next" PC at all? Does it look into the future or not? See other comments on confusion.

Reply:
Zero simply means that a snapshot of the next_trace_pc value wasn't taken and the input should be consulted directly. I'll add a comment.

Reply:
Technically 0 could be a valid value. An app with a jump to PC 0 (with fault and either exit or recovery) is a valid app. Could we have a separate field for whether the PC is valid?